docs/pages/docs/guide/advanced/google-analytics.mdx

```diff
@@ -0,0 +1,174 @@
+import { Steps } from 'nextra/components'
+
+# Google Analytics
+
+This page describes how to configure your Nextra website with Google Analytics (`gtag.js`).
+
+**Note:** This guide assumes you already have a [Google Analytics measurement id](https://support.google.com/analytics/answer/9304153) for your Nextra site.
+
+## Configuration
+
+<Steps>
+
+### Create Logic for Sending Pageview and Events to Google Analytics
+
+Create a `lib/gtag.js` file that exports the following:
+
+```js filename="lib/gtag.js"
+export const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID
+
+export const pageview = url => {
+  window.gtag('config', GA_TRACKING_ID, {
+    page_path: url
+  })
+}
+
+export const event = ({ action, category, label, value }) => {
+  window.gtag('event', action, {
+    event_category: category,
+    event_label: label,
+    value: value
+  })
+}
+```
```

Comment:
personally would recommend this being ts since the rest of the project is
```suggestion
```js filename="lib/gtag.ts"
export const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID

export const pageview = (url: string) => {
  window.gtag('config', GA_TRACKING_ID, {
    page_path: url
  })
}

type EventParams = {
  event_category: string;
  event_label?: string;
  value?: number;
}

export const event = ({ action, params }: { action: string; params: EventParams }) => {
  window.gtag("event", action, params);
};
```
```